### PR TITLE
GOCART2G

### PR DESCRIPTION
**Pull request by WilliamJamieson**
Moved all of GOCART2G from the `GEOSchem_GridComp` and `GMAO_Shared` into this repository.

### CODE REVIEW

File: ESMF/Aerosol_GridComp/Aerosol_StateSpecs.rc [hunk 1]
```diff
@@ -7,62 +7,64 @@ category: IMPORT
...
+#LWI        | 1      | xy   | N   |      | land-ocean-ice_mask
+#U10M       | m s-1  | xy   | N   |      | 10-meter_eastward_wind
+#V10M       | m s-1  | xy   | N   |      | 10-meter_northward_wind
+#AREA       | m^2    | xy   | N   |      | agrid_cell_area
+#USTAR      | m s-1  | xy   | N   |      | equivalent_neutral_10-meter_northward_wind
+#CN_PRCP    |kg/m^2/s| xy   | N   |      | surface_conv._rain_flux_needed_by_land
+#NCN_PRCP   |kg/m^2/s| xy   | N   |      | Non-convective precipitation
+#ZPBL       | m      | xy   | N   |      | planetary_boundary_layer_height
+#SH         | w m-2  | xy   | N   |      | sensible_heat_flux_from_turbulence
+#Z0H        | m      | xy   | N   |      | surface_roughness_for_heat
```
Comment (tclune) [inline ESMF/Aerosol_GridComp/Aerosol_StateSpecs.rc:23]: Why are these commented?   Do we have a different way to get these quantities now?

File: ESMF/Apps/esm.F90 [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        call NUOPC_CompAttributeSet(child, name="Verbosity", value="low", rc=rc)
+        VERIFY_NUOPC_(rc)
+
+        ! Disabling the following macro, e.g. renaming to WITHCONNECTORS_disable,
+        ! will result in a driver that does not call connectors between the model
+        ! components. This mode can be used if all model components are driven 
+        ! as independent models. However, even for independent models the
+        ! connectors can be set here, but will turn into no-ops.
+#define WITHCONNECTORS
+#ifdef WITHCONNECTORS
```
Comment (tclune) [inline ESMF/Apps/esm.F90:114]: We generally try to avoid CPP conditionals in GEOS software.  Are these essential?

File: ESMF/Apps/esmApp.F90 [hunk 3]
```diff
@@ -0,0 +1,80 @@
+!==============================================================================
```
Comment (tclune) [inline ESMF/Apps/esmApp.F90:1]: Why is this file here?   It does not seem to use any GEOS code.

File: ESMF/Apps/experiment/logging.yaml [hunk 4]
```diff
@@ -0,0 +1,108 @@
+schema_version: 1
```
Comment (tclune) [inline ESMF/Apps/experiment/logging.yaml:1]: What is the ./Apps/experiment directory for?

File: ESMF/Aerosol_GridComp/Aerosol_StateSpecs.rc (hunk 1 above)
Comment (WilliamJamieson) [reply ESMF/Aerosol_GridComp/Aerosol_StateSpecs.rc:23]: I'm just using this a template for now.

File: ESMF/Apps/ocn.F90 [hunk 5]
```diff
@@ -0,0 +1,324 @@
+!==============================================================================
```
Comment (tclune) [inline ESMF/Apps/ocn.F90:1]: Why is this file here?

File: ESMF/Apps/esm.F90 (hunk 2 above)
Comment (WilliamJamieson) [reply ESMF/Apps/esm.F90:114]: Will remove

File: ESMF/Apps/esmApp.F90 (hunk 3 above)
Comment (WilliamJamieson) [reply ESMF/Apps/esmApp.F90:1]: Will remove

File: components.yaml [hunk 6]
```diff
@@ -24,11 +24,10 @@ NCEP_Shared:
 GMAO_Shared:
   local: ./ESMF/Shared/GMAO_Shared@
   remote: ../GMAO_Shared.git
-  tag: v1.1.4
   sparse: ./config/GMAO_Shared.sparse
-  develop: master
+  branch: feature/wjamieson/GOCARTrefactor
```
Comment (tclune) [inline components.yaml:28]: We need to tag this and reference a tag.   Probably should issue a release.  Let's discuss with @mathomp4.

File: components.yaml [hunk 7]
```diff
@@ -12,7 +12,7 @@ ecbuild:
 HEMCO:
   local: ./ESMF/HEMCO_GridComp@
   remote: ../HEMCO.git
-  tag: GEOS_12.8
+  branch: bugfix/GOCART_Build_error
```
Comment (tclune) [inline components.yaml:15]: Definitely need a tag if we are going to use HEMCO here.

**Review by tclune — CHANGES_REQUESTED** [review]
Misc. requests sprinkled in as comments.

File: ESMF/Aerosol_GridComp/Aerosol_StateSpecs.rc (hunk 1 above)
Comment (tclune) [reply ESMF/Aerosol_GridComp/Aerosol_StateSpecs.rc:23]: These fields will eventually be needed, so they remain in the spec (but commented).

File: components.yaml (hunk 7 above)
Comment (tclune) [reply components.yaml:15]: Going to deactivate HEMCO for now, as it is not currently used.

File: components.yaml (hunk 6 above)
Comment (tclune) [reply components.yaml:28]: main on GMAO_Shared may now work just fine.    @WilliamJamieson will test.

**Review by tclune — APPROVED** [review]
